Cloudinit coverage

richtja opened the pull request:
This is a set of tests for `avocado.utils.cloudinit` utility, with fixes found on the way, to prepare it for migration to Autils and increase its coverage.

clebergnu reviewed (changes requested):
Hi @richtja, it's great to see this utility coverage increasing and the same time marking its time to be migrated to autils.

I have a few questions and points that you can answer/address.

Thanks!

richtja commented on avocado/utils/cloudinit.py:
```diff
@@ -130,6 +130,8 @@ def iso(
     if username:
         userdata += USERNAME_TEMPLATE.format(username)
+        if isinstance(username, bytes):
+            username = username.decode(astring.ENCODING)
```
The docstring doesn't specify the type of `username` variable, and the selftests uses [bytes](https://github.com/avocado-framework/avocado/blob/master/selftests/unit/utils/cloudinit.py#L34). Therefore, I have presumed that string and bytes are supported types. But if user uses bytes, then the comparison `if username == "root":` won't work therefore I did this change.

Another solution would be to enforce the parameters to be strings and update the selftests, but I was afraid that I will change API by that.

richtja commented on avocado/utils/cloudinit.py:
```diff
@@ -156,6 +156,7 @@ def do_POST(self):
         if path == self.server.instance_id:
             self.server.instance_phoned_back = True
         self.send_response(200)
+        self.end_headers()
```
Sure I will update the commit message.

richtja commented:
Hi @clebergnu, I updated this PR, based on our discussion. Please take a look. I have kept the unused port obtain since I don't have a better solution, but If you have some idea, I don't mind updating it.

clebergnu commented:
> Hi @clebergnu, I updated this PR, based on our discussion. Please take a look. I have kept the unused port obtain since I don't have a better solution, but If you have some idea, I don't mind updating it.

Hi @richtja,

Let's check how much coverage is lost if we remove this test, which altough seldomly, can still cause false positives (if the post gets assigned by others between the probe and actual use).

It may be better to have a slightly smaller coverage and not have a "technical debt" of sorts.

richtja commented:
> > Hi @clebergnu, I updated this PR, based on our discussion. Please take a look. I have kept the unused port obtain since I don't have a better solution, but If you have some idea, I don't mind updating it.
> 
> Hi @richtja,
> 
> Let's check how much coverage is lost if we remove this test, which altough seldomly, can still cause false positives (if the post gets assigned by others between the probe and actual use).
> 
> It may be better to have a slightly smaller coverage and not have a "technical debt" of sorts.

Hi @clebergnu, since it is just two lines of code which is being tested I have removed the test.

clebergnu reviewed (approved):
LGTM, thanks!
